app/src/androidTest/java/org/kiwix/kiwixmobile/ObjectBoxToRoomMigratorTest.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2023 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile
+
+import android.content.Context
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.mockk.mockk
+import io.objectbox.BoxStore
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.jupiter.api.Test
+import org.junit.runner.RunWith
+import org.kiwix.kiwixmobile.core.dao.entities.RecentSearchEntity
+import org.kiwix.kiwixmobile.core.data.KiwixRoomDatabase
+import org.kiwix.kiwixmobile.core.data.remote.ObjectBoxToRoomMigrator
+
+@RunWith(AndroidJUnit4::class)
+class ObjectBoxToRoomMigratorTest {
+
+  private lateinit var context: Context
+  private lateinit var kiwixRoomDatabase: KiwixRoomDatabase
+  private var boxStore: BoxStore = mockk()
+  private lateinit var objectBoxToRoomMigrator: ObjectBoxToRoomMigrator
+
+  @Before
+  fun setup() {
+    context = ApplicationProvider.getApplicationContext()
+    kiwixRoomDatabase = Room.inMemoryDatabaseBuilder(context, KiwixRoomDatabase::class.java)
+      .allowMainThreadQueries()
+      .build()
+    objectBoxToRoomMigrator = ObjectBoxToRoomMigrator()
+    objectBoxToRoomMigrator.kiwixRoomDatabase = kiwixRoomDatabase
+    objectBoxToRoomMigrator.boxStore = boxStore
+  }
+
+  @After
+  fun cleanup() {
+    kiwixRoomDatabase.close()
+    boxStore.close()
+  }
+
+  @Test
+  fun migrateRecentSearch_shouldInsertDataIntoRoomDatabase() = runBlocking {
+    val box = boxStore.boxFor(RecentSearchEntity::class.java)
+    val expectedSearchTerm = "test search"
```

Comment:
Please test edge case, this a feedback I had already given to previous PR. Please read the comments there.